File: app/src/main/java/com/ohmcoe/vultr/ServerAdapter.java
package com.ohmcoe.vultr;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

/**
 * Created by OHM on 10/3/2560.
 */

public class ServerAdapter extends ArrayAdapter<Server> {

    public ServerAdapter(Context context, int resource) {
        super(context, resource);
    }


    public ServerAdapter(Context context, int resource, List<Server> objects) {
        super(context, resource, objects);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = convertView;

        Server server = getItem(position);

        if (view == null)
        {
            LayoutInflater layoutInflater;
            layoutInflater = LayoutInflater.from(getContext());
            view = layoutInflater.inflate(R.layout.server_list, null);
        }

        if (server != null)
        {
            TextView txtServerName = (TextView)view.findViewById(R.id.txtServerName);
            TextView txtPendingCharges = (TextView)view.findViewById(R.id.txtPendingCharges);
            TextView txtIP = (TextView)view.findViewById(R.id.txtIP);

            if (txtServerName != null)
            {
                txtServerName.setText(server.getLabel());
            }

            if (txtPendingCharges != null)
            {
                String pendingCharges = "$" + server.getPending_charges();
                txtPendingCharges.setText(pendingCharges);
            }

            if (txtIP != null)
            {
                txtIP.setText(server.getMain_ip());
            }
        }

        return view;
    }
}

File: app/src/main/java/com/ohmcoe/vultr/ServerAdapter.kt
package com.ohmcoe.vultr

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.ohmcoe.vultr.R.id.txtPendingCharges
import com.ohmcoe.vultr.R.id.txtServerName
import kotlinx.android.synthetic.main.server_list.view.*

class ServerAdapter : ArrayAdapter<Server> {

    constructor(context: Context, resource: Int) : super(context, resource) {}

    constructor(context: Context, resource: Int, objects: List<Server>) : super(context, resource, objects) {}

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView

        val server = getItem(position)

        if (view == null) {
            val layoutInflater: LayoutInflater
            layoutInflater = LayoutInflater.from(context)
            view = layoutInflater.inflate(R.layout.server_list, null)
        }

        if (server != null) {
            val txtServerName = view!!.txtServerName
            val txtPendingCharges = view.txtPendingCharges;
            val txtIP = view.txtIP

            if (txtServerName != null) {
                txtServerName.text = server.label
            }

            if (txtPendingCharges != null) {
                val pendingCharges = "$" + server.pending_charges!!
                txtPendingCharges.text = pendingCharges
            }

            if (txtIP != null) {
                txtIP.text = server.main_ip
            }
        }

        return view!!
    }
}
